### PR TITLE
Enable dependency verification

### PR DESCRIPTION
**Pull request by marcphilipp**
## Overview

In order to avoid potential supply chain attacks, this commit enables Gradle's [dependency verification](https://docs.gradle.org/current/userguide/dependency_verification.html) features and enables signature and checksum verification.

The file was generated using the following command:

    ./gradlew --write-verification-metadata pgp,sha512 \
              spotlessCheck compileTestGroovy

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](https://junit.org/junit5/docs/snapshot/api/org.junit.platform.commons/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [x] Change is documented in the [User Guide](https://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](https://junit.org/junit5/docs/snapshot/user-guide/#release-notes)
- [ ] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

File: gradle/verification-metadata.xml [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+            <sha512 value="0f361d125c297761706232f08b08886fb6a03e993ea726dba9accff506b3818dc4416f6cd06005ef223503a952b2e8b3024c6ca921f74e463ccce56016379b0f" origin="Generated by Gradle because artifact wasn't signed"/>
+         </artifact>
+      </component>
+      <component group="de.undercouch" name="gradle-download-task" version="4.1.1">
+         <artifact name="gradle-download-task-4.1.1.jar">
+            <pgp value="1fa37fbe4453c1073e7ef61d6449005f96bc97a3"/>
+         </artifact>
+         <artifact name="gradle-download-task-4.1.1.pom">
+            <ignored-keys>
+               <ignored-key id="1fa37fbe4453c1073e7ef61d6449005f96bc97a3" reason="PGP verification failed"/>
```
Comment (marcphilipp) [inline gradle/verification-metadata.xml:168]: ☝️ to be investigated

File: gradle/verification-metadata.xml [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+      </component>
+      <component group="org.jacoco" name="org.jacoco.build" version="0.8.7">
+         <artifact name="org.jacoco.build-0.8.7.pom">
+            <sha512 value="c675dc20d3d192a4193d651a6fa3ddac3bfe97844be536146ca6e78c29c1559b06fe9495be39d4dbf606b8a2cb0720391a6fe53f37d628949659c3224e4eaa8d" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
+      <component group="org.jetbrains.kotlin" name="kotlin-gradle-plugin" version="1.7.10">
+         <artifact name="kotlin-gradle-plugin-1.7.10.module">
+            <ignored-keys>
+               <ignored-key id="6f538074ccebf35f28af9b066a0975f8b1127b83" reason="PGP verification failed"/>
```
Comment (marcphilipp) [inline gradle/verification-metadata.xml:340]: ☝️ to be investigated

File: gradle/verification-metadata.xml [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+      <component group="org.ysb33r.gradle" name="grolifant" version="0.16.1">
+         <artifact name="grolifant-0.16.1.jar">
+            <ignored-keys>
+               <ignored-key id="ea022560a81e5bd48db3d18b54ac8e2d98cfeac6" reason="PGP verification failed"/>
+            </ignored-keys>
+            <sha512 value="c4f201d3c1243f701751e0ebec32f15fe957c362d6d5a70eec7569e938e55f30b5f60965807a3e2d4bb61526648903f3c88a74f6bccde21da1e7b45e7a17dfa1" origin="Generated by Gradle because PGP signature verification failed!"/>
+         </artifact>
+         <artifact name="grolifant-0.16.1.pom">
+            <ignored-keys>
+               <ignored-key id="ea022560a81e5bd48db3d18b54ac8e2d98cfeac6" reason="PGP verification failed"/>
```
Comment (marcphilipp) [inline gradle/verification-metadata.xml:384]: ☝️ to be investigated

**Comment by jbduncan** [comment]
@marcphilipp I get the impression this PR is important enough to re-open. WDYT?
